compiler/src/iree/compiler/DispatchCreation/FusionUtils.cpp

```diff
@@ -44,13 +44,15 @@ bool areFusableAsElementwiseOps(MLIRContext *context, OpOperand *fusedOperand,
   }
 
   // If the generic op is "just" copy, then fuse always.
-  Block &body = producerOp->getRegion(0).front();
-  if (std::begin(body)->hasTrait<OpTrait::IsTerminator>())
-    return true;
-  if (llvm::all_of(body.getArguments(),
-                   [](BlockArgument arg) { return arg.use_empty(); })) {
-    // The operands aren't used, its just an `linalg.index` op.
-    return true;
+  if (producerOp->getNumRegions() != 0) {
```

Comment:
WHy did we need this check?

Reply:
Because I was using `fuseElementwiseOpsControlFn` (in ElementwiseOpFusion.cpp) as a control function. `fuseElementwiseOpsControlFn` ends up calling `areFusableElementwiseOps` with the `AttentionOp` as the producer. This usage might be incorrect, however. I was attempting to keep the logic the same, but I think the vast majority doesn't apply to transpose folding, the only important part is ensuring that the producer & consumer are not in dispatches.

So, I created a separate control function for this. And reverted the num regions check